frontend/src/utils/urlHyperlinkInserter.ts

```diff
@@ -0,0 +1,46 @@
+import { EditorView } from "codemirror";
+import { CodePairDocType } from "../store/editorSlice";
+
+const isValidUrl = (url: string) => {
+	// eslint-disable-next-line no-useless-escape
+	const urlRegex = /^(https?|ftp):\/\/(-\.)?([^\s\/?\.#-]+\.?)+(\/[^\s]*)?$/i;
+	return urlRegex.test(url);
+};
```

Comment:
How about switching to [`validator`](https://github.com/validatorjs/validator.js)? I don’t want to manage regular expression code in CodePair.